python/pyarrow/tests/test_adhoc_memory_leak.py

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pytest
+
+import numpy as np
+import pyarrow as pa
+
+import pyarrow.tests.util as test_util
+
+try:
+    import pandas as pd
+except ImportError:
+    pass
+
+
+@pytest.mark.memory_leak
+@pytest.mark.pandas
+def test_deserialize_pandas_arrow_7956():
+    df = pd.DataFrame({'a': np.arange(10000),
+                       'b': [pd.util.testing.rands(5) for _ in range(10000)]})
```

Comment:
Future note: we should not use pandas.util.testing, as that is deprecated (my bad for using it in the reproducer example)

Reply:
Ah thanks for letting me know. Looks like we have a couple lingering usages

```
python/pyarrow/tests/test_adhoc_memory_leak.py
35:                       'b': [pd.util.testing.rands(5) for _ in range(10000)]})

python/pyarrow/tests/test_plasma.py
411:        pd.util.testing.assert_frame_equal(df, result)
```

Reply:
Yes, I already cleaned up most some time ago, but clearly missed the plasma one.